frontend/src/components/filterResultsModal/filterResultsModal.tsx

```diff
@@ -157,20 +176,30 @@ export const FilterResultsModal = (props: FilterResultsModalProps): JSX.Element
 
                         {currentMenu === "Venue" && (
                             <>
-                                <label>Venue</label>
-                                <Form.Control
-                                    type="text"
-                                    className="venue-input"
-                                    placeholder="Enter venue"
-                                    value={filters.location ? filters.location : ""}
-                                    // Controlled with the current location value
-                                    onChange={(e) => updateFilters({ ...filters, location: e.target.value })}
-                                />
+                                <label htmlFor="venue-select">Venue</label>
+                                <select
+                                    id="venue-select"
+                                    className="venue-input venue-filters"
+                                    value={selectedVenue}
+                                    onChange={(e) => {
+                                        const newSelectedVenue = e.target.value;
+                                        setSelectedVenue(newSelectedVenue);
+                                        updateFilters({ ...filters, location: newSelectedVenue }); // Update filters with selected venue
+                                    }}
+                                >
+                                    <option value="">Select venue</option>
+                                    {AllLocations.map((location, index) => (
```

Comment:
While unlikely, if there are no locations, this will error out and likely result in the page not being rendered, you can do a check prior to rendering

```javascript
<option value="">Select venue</option>
{allLocations.length === 0 ? (
      <option disabled>No venues found</option>
) : (
allLocations.map((location, index) => (
     <option key={index} value={location.address}>
         {location.address}
     </option>
         )
       )
    )}  
```